stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1245,36 +1253,54 @@ void DiscardingTaskGroup::offer(AsyncTask *completedTask, AsyncContext *context)
         switch (readyErrorItem.getStatus()) {
           case ReadyStatus::RawError:
             SWIFT_TASK_GROUP_DEBUG_LOG(this, "offer, complete, resume with raw error:%p", readyErrorItem.getRawError(this));
-            resumeWaitingTaskWithError(readyErrorItem.getRawError(this), assumed,
+            // The following MUST be done in the following order: detach, unlock, resume waitingTask.
+            // because we do not want to allow another task to run and have the potential to lock or even destroy
+            // the group before we've given up the lock.
+            _swift_taskGroup_detachChild(asAbstract(this), completedTask);
+            unlock();
+            return resumeWaitingTaskWithError(readyErrorItem.getRawError(this), assumed,
```

Comment:
Same issue as https://github.com/apple/swift/pull/67700 fixed, but inside the offer impl